ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py

```diff
@@ -490,20 +526,40 @@ def _router_bgp_vlan_aware_bundles(self) -> list | None:
                     bundle["vlan"] = list_compress([int(l2vlan["id"]) for l2vlan in l2vlans])
                     bundle = {"name": bundle_name, **bundle}
 
+                    append_if_not_duplicate(
+                        list_of_dicts=bundles,
+                        primary_key="name",
+                        new_dict=bundle,
+                        context=(
+                            "BGP VLAN-Aware Bundles defined under network services. A common reason is that an 'l2vlan' name overlaps with an"
+                            " 'evpn_vlan_bundle' name"
+                        ),
+                        context_keys=["name"],
+                    )
+
+            # L2VLANs and SVIs which have an evpn_vlan_bundle defined
+            for bundle_name, bundle_dict in l2vlan_svi_vlan_aware_bundles.items():
+                evpn_vlan_bundle = bundle_dict["evpn_vlan_bundle"]
+                l2vlans_svis = bundle_dict["l2vlan_svis"]
```

Comment:
I think it would be simpler to maintain and understand if we had one dict of bundles to create like `{<bundle_name>: [<vlan number>, <vlan number> ...]}` which is populated during parsing of all vlans and svis. Then at the end here you can just parse that dict a create the bundles.